pulser-core/pulser/backend/config.py

```diff
@@ -14,28 +14,223 @@
+        cls_name = self.__class__.__name__
+        if invalid_kwargs := (
+            set(backend_options)
+            - (self._expected_kwargs() | {"backend_options"})
+        ):
+            warnings.warn(
+                f"{cls_name!r} received unexpected keyword arguments: "
+                f"{invalid_kwargs}; only the following keyword "
+                f"arguments are expected: {self._expected_kwargs()}.",
+                stacklevel=2,
+            )
+        # Prevents potential issues with mutable arguments
+        self._backend_options = copy.deepcopy(backend_options)
+        if "backend_options" in backend_options:
+            with warnings.catch_warnings():
+                warnings.filterwarnings("always")
+                warnings.warn(
+                    f"The 'backend_options' argument of {cls_name!r} "
+                    "has been deprecated. Please provide the options "
+                    f"as keyword arguments directly to '{cls_name}()'.",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
+            self._backend_options.update(backend_options["backend_options"])
+
+    def _expected_kwargs(self) -> set[str]:
+        return set()
+
+    def __getattr__(self, name: str) -> Any:
+        if (
+            # Needed to avoid recursion error
+            "_backend_options" in self.__dict__
+            and name in self._backend_options
+        ):
+            return self._backend_options[name]
+        raise AttributeError(f"{name!r} has not been passed to {self!r}.")
+
+
+class EmulationConfig(BackendConfig, Generic[StateType]):
+    """Configures an emulation on a backend.
+
+    Args:
+        observables: A sequence of observables to compute at specific
+            evaluation times. The observables without specified evaluation
+            times will use this configuration's 'default_evaluation_times'.
+        default_evaluation_times: The default times at which observables
+            are computed. Can be a sequence of relative times between 0
+            (the start of the sequence) and 1 (the end of the sequence).
+            Can also be specified as "Full", in which case every step in the
+            emulation will also be an evaluation times.
+        initial_state: The initial state from which emulation starts. If
+            specified, the state type needs to be compatible with the emulator
+            backend. If left undefined, defaults to starting with all qudits
+            in the ground state.
+        with_modulation: Whether to emulate the sequence with the programmed
+            input or the expected output.
+        interaction_matrix: An optional interaction matrix to replace the
+            interaction terms in the Hamiltonian. For an N-qudit system,
+            must be an NxN symmetric matrix where entry (i, j) dictates
+            the interaction coefficient between qudits i and j, ie it replaces
+            the C/r_{ij}^6.
```

Comment:
c_n/r{ij}^n? There is also XY.

Reply:
Pablo recently added a `log_file` to our `BackendConfig` it would cause the application to pipe all output into the log file. It seems like a thing we could generally support, though I'm also fine with keeping it an emu-mps specific config value.

Reply:
It sounds a bit EMU-MPS specific at this moment. I'd keep it there for now, we can always extend it to `EmulationConfig` if it makes sense in the future